File: scala/src/main/scala/introduction.scala
import com.jambit.practicalresult.*

@main def introduction: Unit =
  val result: Either[String, List[Int]] = listService(4)

  val msg: String = result match {
    case Right(list) => list.mkString("-")
    case Left(err) => err
  }
  
  println(msg)

File: kotlin/src/main/java/Introduction.kt
import arrow.core.Either

fun main() {
  val result: Either<String, List<Int>> = listService(4)

  val msg: String = when (result) {
    is Either.Right -> result.value.joinToString("-")
    is Either.Left  -> result.value
  }

  println(msg)
}